pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+
+
+class FakedWBEMConnection(WBEMConnection):
+    """
+    A faked WBEMConnection class that mocks _imethodcall and _methodcall
+    WBEMConnection methods and creates equivalents for all of
+    the WBEMConnection methods that communicate with the server.
+    Because it only mocks _imethodcall and _methodcall all of the logic in
+    the various WBEMConneciton methods that prepare requests for the server
+    (ex. GetClass) is utilized and the parameters of the method calls are
+    tested/modified, etc.
+    """
+    def __init__(self, verbose=None):
+        super(FakedWBEMConnection, self).__init__(
+            'http://FakedUrl', default_namespace=DEFAULT_NAMESPACE,
+            # creds=None,
+            # x509=None, verify_callback=None, ca_certs=None,
+            # no_verifi cation=False, timeout=None,
+            use_pull_operations=False,
+            enable_stats=False)
+
+        # TODO we are not useing NocaseDict.  Are namespace names case
+        # independent?????
+        # Dictionary of classes representing a repository.  This is
+        # a dictionary of dictionaries where the top level key is
+        # namespace and the keys for each dictionary in a namespace are
+        # class names in the repository. The values in each subdirectory are
+        # CIMClass objects.
+        self.classes = {}
+
+        # Same format as classes above, dictionary of dictionaries where
+        # the top level keys are namespaces and the next level kesy are
+        # CIMQualifierDeclaration names with values of the
+        # CIMQualifierDeclaration objects
+        self.qualifiers = {}
+
+        # Top level dictionary: key is namespace,
+        # and value is list of instances in that namespace
+        # TODO we should really have a subdict per class.
+        self.instances = {}
+
+        # Open Pull Contexts. The key for each context is an enumeration
+        # context id.  The data is the total list of instances/names to
+        # be returned and the current position in the list. Any context in
+        # this list is still open.
+        self.enumeration_contexts = {}
+
+        self.verbose = verbose
+
+        self.logfile = None
+
+        self._imethodcall = Mock(side_effect=self._mock_imethodcall)
+        self._methodcall = Mock(side_effect=self._mock_methodcall)
+
+    ##########################################################
+    #
+    #   Functions Mocked
+    #
+    ##########################################################
+
+    def _mock_imethodcall(self, methodname, namespace, response_params_rqd=None,
+                          **params):
+        """
+        Mock the WBEMConnection _imethod call.
+
+        This mock calls functions within this class that fake the processing
+        in a WBEM server (at the CIM Object level) for the varisous CIM/XML
+        methods and return.
+
+        Each function is named with the lower case method namd prepended with
+        'fake_'.
```

Comment:
We prepend with `_fake_` not with `fake_`.

Reply:
Done